pkg/api/validation/schema.go

```diff
@@ -62,6 +64,70 @@ type NullSchema struct{}
 
 func (NullSchema) ValidateBytes(data []byte) error { return nil }
 
+type NoDoubleKeySchema struct{}
+
+func (NoDoubleKeySchema) ValidateBytes(data []byte) error {
```

Comment:
the trouble is that this is bytes, we could automatically build up a map for every runtime.Object I suppose.  Follow on PR?